docs/threshold_encryption.md

```diff
@@ -0,0 +1,41 @@
+# Threshold Encryption
+*Threshold encryption* is one of cryptosystem, if in order to decrypt an encrypted message or to sign a message, several 
+parties (more than some threshold number) must cooperate in the decryption or signature protocol.
+The message is encrypted using a `public key` and the corresponding `private key` is shared among the participating parties.
+Only by possessing more than a threshold number of bits of the secret can the secret be determined.
+
+### Idea
+The idea of threshold cryptography is to protect information (or computation) by fault-tolerance distributing it among a
+cluster of cooperating computers. First consider the fundamental problem of threshold cryptography, a problem of secure
+sharing of a secret. A secret sharing scheme allows one to distribute a piece of secret information among several servers
+in a way that meets the following requirements.
+1. No group of corrupt servers (smaller than a given threshold) can figure out what the secret is, even if they cooperate.
+2. When it becomes necessary that the secret information be reconstructed, a large enough number of servers (a number
+larger than the above threshold) can always do it.
+
+### Example
+If I had four copies of a key and cut each key into three pieces, distributing one piece each to twelve people then it 
```

Comment:
* Original threshold encryption algorithm generates copies of a key? 
* And is this the mechanism that TE makes threshold?  

Since I have no background knowledge about TE, this sentence is hard to understand